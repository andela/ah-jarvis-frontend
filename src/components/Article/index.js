import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import AuthorDetails from '../AuthorDetails';
import thumbsUp from '../../assets/icons/thumbs-up.svg';
import heart from '../../assets/icons/heart.svg';

const Article = ({
  title, author, image, preview, date, slug, readtime, likesCount,
}) => (
  <div className="col m11 s12 preview">
    <div className="row author">
      <AuthorDetails user={author} date={date} small readtime={readtime} slug={slug} />
    </div>
    {image && (
      <Link to={`article/${slug}`} className="black-text">
        <div className="row article__image">
          <img alt="" src={image} className="responsive-img article__preview" />
        </div>
      </Link>
    )}
    <div className="row article__details">
      <Link to={`article/${slug}`} className="black-text">
        <h2 className="article__preview--title truncate">{title}</h2>
      </Link>
      <div className="article__preview--body truncate">
        <Link to={`article/${slug}`} className="black-text">
          <p>{preview}</p>
        </Link>
        <Link to={`article/${slug}`}>Read more...</Link>
      </div>
    </div>
    <div className="row">
      <div className="article__hypes">
        <div className="icon---default">
          <img src={thumbsUp} alt="thumbs-up" />
          <div className="text--small">{likesCount}</div>
        </div>
        <div>
          <img src={heart} alt="thumbs-up" />
        </div>
      </div>
    </div>
  </div>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  author: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
  readtime: PropTypes.number,
  likesCount: PropTypes.string.isRequired,
};

export default Article;
