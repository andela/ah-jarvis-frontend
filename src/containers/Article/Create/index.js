import React, { Component } from 'react';
import Dante from 'Dante2';
import M from 'materialize-css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ImageBlockConfig } from 'Dante2/package/es/components/blocks/image';
import { VideoBlockConfig } from 'Dante2/package/es/components/blocks/video';
import { EmbedBlockConfig } from 'Dante2/package/es/components/blocks/embed';
import { PlaceholderBlockConfig } from 'Dante2/package/es/components/blocks/placeholder';

import createArticleAction from './actions';
import UserInfo from '../../../components/UserInfo';
import Header from '../../../components/Header';
import editorstate from '../../../utils/editorstate';
import config from '../../../config';
import uploader from '../../../utils/uploader';

class Create extends Component {
  state = {
    saving: false,
    tags: [],
  };

  getTags = (_e, d) => {
    this.setState({
      tags: [...this.state.tags, d.childNodes[0].nodeValue],
    });
  };

  removeTag = (_e, d) => {
    this.setState({
      tags: this.state.tags.filter(t => t !== d.childNodes[0].nodeValue),
    });
  };

  handleSave = (state) => {
    this.setState({ saving: true });
    const editorState = state.editorState();
    const title = editorState.getCurrentContent().getFirstBlock().text;
    const data = {
      article: {
        body: JSON.stringify(state.editorContent),
        title,
        description: title,
      },
    };

    localStorage.setItem('article', JSON.stringify(data));
    setTimeout(() => {
      this.setState({ saving: false });
    }, 500);
  };

  handlePublish = () => {
    const { postArticle, history } = this.props;
    const article = JSON.parse(localStorage.getItem('article'));
    article.article.tagList = this.state.tags;
    postArticle(article, history);
    localStorage.removeItem('article');
  };

  handleFailure = (e) => {
    M.toast({ html: e, classes: 'danger' });
  };

  handleUpload = (img, state) => {
    uploader({
      body: img,
      progress: (e) => {
        state.updateProgressBar(e);
      },
    }).then((data) => {
      state.uploadCompleted(data.secure_url);
    });
  };

  render() {
    const { publishing } = this.props.article;

    return (
      <React.Fragment>
        <Header {...this.props} />
        <div className="container m-t--30">
          <UserInfo
            onPublish={this.handlePublish}
            publishing={publishing}
            save={this.state.saving}
            user={this.props.user}
            getTags={this.getTags}
            removeTag={this.removeTag}
          />
          <div className="row">
            <div className="col s12">
              <Dante
                content={editorstate}
                spellcheck
                widgets={[
                  ImageBlockConfig({
                    upload_handler: this.handleUpload,
                    options: {
                      upload_url: null,
                      upload_handler: this.handleUpload,
                      upload_callback: this.handleUpload,
                      upload_error_callback: this.handleFailure,
                    },
                  }),
                  VideoBlockConfig(),
                  EmbedBlockConfig(),
                  PlaceholderBlockConfig(),
                ]}
                data_storage={{
                  url: config.BASE_URL,
                  method: 'POST',
                  save_handler: this.handleSave,
                  success_handler: this.handleSuccess,
                  failure_handler: this.handleFailure,
                  interval: 1000,
                }}
              />
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

Create.propTypes = {
  article: PropTypes.shape({
    publishing: PropTypes.bool.isRequired,
  }).isRequired,
  postArticle: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  user: PropTypes.shape({
    username: PropTypes.string,
  }).isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    postArticle: createArticleAction,
  },
  dispatch,
);

const mapStateToProps = ({ createArticle }) => ({
  article: createArticle,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Create);
