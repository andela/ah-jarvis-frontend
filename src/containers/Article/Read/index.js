import Dante from 'Dante2';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import M from 'materialize-css';

import thumbsUp from '../../../assets/icons/thumbsUp.svg';
import thumbsDown from '../../../assets/icons/thumbsDown.svg';
import heart from '../../../assets/icons/baseline-bookmark_border-24px.svg';
import heartFill from '../../../assets/icons/baseline-bookmark-24px.svg';

import Header from '../../../components/Header';
import {
  fetchArticle, rateSuccess, likeArticle, dislikeArticle, bookmarkArticle,
} from './actions';
import Reaction from '../../../components/LikeDislike';
import AuthorDetails from '../../../components/AuthorDetails';
import ArticleDetailsLoader from '../../../components/Placehoders/ArticleDetailsLoader ';
import NotFound from '../../../components/NotFound';
import api from '../../../utils/api';
import getCurrentUser from '../../../utils/auth';
import readTime from '../../../utils/readtime';

class Read extends Component {
  state = {
    rating: 0,
    alert: false,
    alertClass: 'success',
    alertMessage: '',
  };

  componentDidMount() {
    const { getArticle, match } = this.props;
    let auth = false;
    if (getCurrentUser()) {
      auth = true;
    }
    getArticle(match.params.id, auth);
  }

  toaster = () => {
    M.toast({ html: this.state.alertMessage, className: this.state.alertClass });
  };

  onStarClick = (nextValue) => {
    this.setState({ rating: nextValue });
    const rateData = {
      rate: {
        rate: nextValue,
      },
    };

    if (getCurrentUser()) {
      this.setState({ alert: false });
      api({
        endpoint: `/articles/${this.props.match.params.id}/rate/`,
        method: 'POST',
        data: rateData,
        authenticated: true,
      })
        .then(() => {
          const { getRating, getArticle, match } = this.props;
          getRating();
          getArticle(match.params.id);
          console.log('article here', getArticle(match.params.id));
        })
        .catch((err) => {
          this.setState({
            alert: true,
            alertMessage: err.rate.errors.message[0],
            alertClass: 'danger',
          });
        });
    } else {
      this.setState({
        alert: true,
        alertMessage: 'Please Login to rate this article',
        alertClass: 'danger',
      });
    }
  };

  handleReaction = (e) => {
    e.preventDefault();
    const {
      like, dislike, match, bookmark,
    } = this.props;

    if (getCurrentUser()) {
      this.setState({ alert: false });
      if (e.target.id === 'like') {
        like(match.params.id);
      } else if (e.target.id === 'dislike') {
        dislike(match.params.id);
      } else if (e.target.id === 'bookmark') {
        bookmark(match.params.id, 'DELETE');
      } else if (e.target.id === 'unBookmark') {
        bookmark(match.params.id, 'POST');
      }
    } else {
      this.setState({
        alert: true,
        alertMessage: 'Please Login to like or dislike this article',
        alertClass: 'danger',
      });
    }
  };

  renderReaction = (id, src, count) => (
    <Reaction id={id} src={src} count={count} onClick={this.handleReaction} />
  );

  renderTags = tags => tags.map(tag => <div className="chip">{tag}</div>);

  render() {
    const {
      isFetching, success, payload, errors, isRating,
    } = this.props.article;
    let data;
    let readtime;
    if (payload.article) {
      try {
        data = JSON.parse(payload.article.body);
        readtime = readTime(data);
      } catch (e) {
        return <NotFound />;
      }
      console.log(payload.article.favorited);
    }
    if (errors) {
      return <NotFound />;
    }
    return (
      <React.Fragment>
        <Header {...this.props} />

        <div className="container m-t--30">
          <div className="row">
            {this.state.alert ? this.toaster() : ''}
            {(isFetching || !success) && !isRating ? (
              <ArticleDetailsLoader />
            ) : (
              <React.Fragment>
                <div className="col s11">
                  <AuthorDetails
                    user={{ ...payload.article.author }}
                    date={payload.article.created_at}
                    slug={payload.article.slug}
                    readtime={readtime}
                    averageRate={
                      payload.article.average_rating
                        ? parseFloat(payload.article.average_rating)
                        : this.state.rating
                    }
                    onStarClick={this.onStarClick}
                  />
                  <Dante read_only content={data} />
                </div>

                <div className="col s1">
                  <div className="reactions">
                    {this.renderReaction('like', thumbsUp, payload.article.likes_count)}
                    {this.renderReaction('dislike', thumbsDown, payload.article.dislikes_count)}
                    {payload.article.favorited ? this.renderReaction('bookmark', heartFill) : this.renderReaction('unBookmark', heart)}
                  </div>
                </div>

              </React.Fragment>
            )}
          </div>
          {payload.article && <div className="row">{ this.renderTags(payload.article.tagList) }</div>}
`
          {' '}
        </div>
      </React.Fragment>
    );
  }
}

Read.propTypes = {
  article: PropTypes.shape({
    isFetching: PropTypes.bool.isRequired,
    isRating: PropTypes.bool.isRequired,
    success: PropTypes.bool.isRequired,
    payload: PropTypes.object.isRequired,
    errors: PropTypes.object,
  }).isRequired,
  getArticle: PropTypes.func.isRequired,
  getRating: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.object.isRequired,
  }).isRequired,
  like: PropTypes.func.isRequired,
  dislike: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    getArticle: fetchArticle,
    getRating: rateSuccess,
    like: likeArticle,
    dislike: dislikeArticle,
    bookmark: bookmarkArticle,
  },
  dispatch,
);

const mapStateToProps = state => ({
  article: state.fetchArticle,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Read);
