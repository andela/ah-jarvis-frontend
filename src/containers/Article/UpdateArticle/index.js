import React, { Component } from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Header from '../../../components/Header';
import Editor from '../../../components/Editor';
import { fetchArticle } from '../Read/actions';
import createArticleAction from '../Create/actions';

class Update extends Component {
  componentDidMount() {
    this.props.getArticle(this.props.match.params.slug);
  }

  edit = (article, history) => {
    this.props.editArticle(article, history, true, this.props.match.params.slug);
  };

  render() {
    const {
      isFetching, payload, success, publishing,
    } = this.props.article;
    const { slug } = this.props.match.params;
    return (
      <React.Fragment>
        <Header />
        {success
          && !isFetching && (
            <Editor
              state={JSON.parse(payload.article.body)}
              postArticle={this.edit}
              slug={slug}
              history={this.props.history}
              publishing={publishing}
              user={this.props.user}
              tags={payload.article.tagList}
              update
            />
        )}
      </React.Fragment>
    );
  }
}

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    getArticle: fetchArticle,
    editArticle: createArticleAction,
  },
  dispatch,
);

const mapStateToProps = state => ({
  article: state.fetchArticle,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Update);

Update.propTypes = {
  article: PropTypes.object.isRequired,
  editArticle: PropTypes.func.isRequired,
  getArticle: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.object,
  user: PropTypes.shape({
    username: PropTypes.string,
  }).isRequired,
  slug: PropTypes.string.isRequired,
};
