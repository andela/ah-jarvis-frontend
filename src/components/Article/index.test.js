import { shallow } from 'enzyme';
import React from 'react';
import Article from '.';

describe('<Article />', () => {
  it('renders  <Article /> components', () => {
    const snap = shallow(<Article author={{ username: 'test user' }} date="" image="" slug="data" preview="data" />);
    expect(snap).toMatchSnapshot();
  });
});
