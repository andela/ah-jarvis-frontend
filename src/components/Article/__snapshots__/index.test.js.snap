// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> renders  <Article /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    author={
        Object {
            "username": "test user",
          }
    }
    date=""
    image=""
    preview="data"
    slug="data"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row author"
>
          <AuthorDetails
                    date=""
                    small={true}
                    user={
                              Object {
                                        "username": "test user",
                                      }
                    }
          />
</div>,
        "",
        <div
          className="row article__details"
>
          <Link
                    className="black-text"
                    replace={false}
                    to="article/data"
          >
                    <h2
                              className="article__preview--title truncate"
                    />
          </Link>
          <div
                    className="article__preview--body truncate"
          >
                    <Link
                              className="black-text"
                              replace={false}
                              to="article/data"
                    >
                              <p>
                                        data
                              </p>
                    </Link>
                    <Link
                              replace={false}
                              to="article/data"
                    >
                              Read more...
                    </Link>
          </div>
</div>,
        <div
          className="row"
>
          <div
                    className="article__hypes"
          >
                    <div
                              className="icon---default"
                    >
                              <img
                                        alt="thumbs-up"
                                        src="thumbs-up.svg"
                              />
                              <div
                                        className="text--small"
                              >
                                        77k
                              </div>
                    </div>
                    <div>
                              <img
                                        alt="thumbs-up"
                                        src="heart.svg"
                              />
                    </div>
          </div>
</div>,
      ],
      "className": "col m11 s12 preview",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <AuthorDetails
            date=""
            small={true}
            user={
                        Object {
                                    "username": "test user",
                                  }
            }
/>,
          "className": "row author",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "date": "",
            "small": true,
            "user": Object {
              "username": "test user",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="black-text"
              replace={false}
              to="article/data"
>
              <h2
                            className="article__preview--title truncate"
              />
</Link>,
            <div
              className="article__preview--body truncate"
>
              <Link
                            className="black-text"
                            replace={false}
                            to="article/data"
              >
                            <p>
                                          data
                            </p>
              </Link>
              <Link
                            replace={false}
                            to="article/data"
              >
                            Read more...
              </Link>
</div>,
          ],
          "className": "row article__details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2
                className="article__preview--title truncate"
/>,
              "className": "black-text",
              "replace": false,
              "to": "article/data",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "article__preview--title truncate",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="black-text"
                  replace={false}
                  to="article/data"
>
                  <p>
                                    data
                  </p>
</Link>,
                <Link
                  replace={false}
                  to="article/data"
>
                  Read more...
</Link>,
              ],
              "className": "article__preview--body truncate",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <p>
                    data
</p>,
                  "className": "black-text",
                  "replace": false,
                  "to": "article/data",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "data",
                  },
                  "ref": null,
                  "rendered": "data",
                  "type": "p",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Read more...",
                  "replace": false,
                  "to": "article/data",
                },
                "ref": null,
                "rendered": "Read more...",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="article__hypes"
>
            <div
                        className="icon---default"
            >
                        <img
                                    alt="thumbs-up"
                                    src="thumbs-up.svg"
                        />
                        <div
                                    className="text--small"
                        >
                                    77k
                        </div>
            </div>
            <div>
                        <img
                                    alt="thumbs-up"
                                    src="heart.svg"
                        />
            </div>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="icon---default"
>
                <img
                                alt="thumbs-up"
                                src="thumbs-up.svg"
                />
                <div
                                className="text--small"
                >
                                77k
                </div>
</div>,
              <div>
                <img
                                alt="thumbs-up"
                                src="heart.svg"
                />
</div>,
            ],
            "className": "article__hypes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="thumbs-up"
                    src="thumbs-up.svg"
/>,
                  <div
                    className="text--small"
>
                    77k
</div>,
                ],
                "className": "icon---default",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "thumbs-up",
                    "src": "thumbs-up.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "77k",
                    "className": "text--small",
                  },
                  "ref": null,
                  "rendered": "77k",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="thumbs-up"
                  src="heart.svg"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "thumbs-up",
                  "src": "heart.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row author"
>
            <AuthorDetails
                        date=""
                        small={true}
                        user={
                                    Object {
                                                "username": "test user",
                                              }
                        }
            />
</div>,
          "",
          <div
            className="row article__details"
>
            <Link
                        className="black-text"
                        replace={false}
                        to="article/data"
            >
                        <h2
                                    className="article__preview--title truncate"
                        />
            </Link>
            <div
                        className="article__preview--body truncate"
            >
                        <Link
                                    className="black-text"
                                    replace={false}
                                    to="article/data"
                        >
                                    <p>
                                                data
                                    </p>
                        </Link>
                        <Link
                                    replace={false}
                                    to="article/data"
                        >
                                    Read more...
                        </Link>
            </div>
</div>,
          <div
            className="row"
>
            <div
                        className="article__hypes"
            >
                        <div
                                    className="icon---default"
                        >
                                    <img
                                                alt="thumbs-up"
                                                src="thumbs-up.svg"
                                    />
                                    <div
                                                className="text--small"
                                    >
                                                77k
                                    </div>
                        </div>
                        <div>
                                    <img
                                                alt="thumbs-up"
                                                src="heart.svg"
                                    />
                        </div>
            </div>
</div>,
        ],
        "className": "col m11 s12 preview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <AuthorDetails
              date=""
              small={true}
              user={
                            Object {
                                          "username": "test user",
                                        }
              }
/>,
            "className": "row author",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "date": "",
              "small": true,
              "user": Object {
                "username": "test user",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="black-text"
                replace={false}
                to="article/data"
>
                <h2
                                className="article__preview--title truncate"
                />
</Link>,
              <div
                className="article__preview--body truncate"
>
                <Link
                                className="black-text"
                                replace={false}
                                to="article/data"
                >
                                <p>
                                                data
                                </p>
                </Link>
                <Link
                                replace={false}
                                to="article/data"
                >
                                Read more...
                </Link>
</div>,
            ],
            "className": "row article__details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h2
                  className="article__preview--title truncate"
/>,
                "className": "black-text",
                "replace": false,
                "to": "article/data",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "article__preview--title truncate",
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="black-text"
                    replace={false}
                    to="article/data"
>
                    <p>
                                        data
                    </p>
</Link>,
                  <Link
                    replace={false}
                    to="article/data"
>
                    Read more...
</Link>,
                ],
                "className": "article__preview--body truncate",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <p>
                      data
</p>,
                    "className": "black-text",
                    "replace": false,
                    "to": "article/data",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "data",
                    },
                    "ref": null,
                    "rendered": "data",
                    "type": "p",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Read more...",
                    "replace": false,
                    "to": "article/data",
                  },
                  "ref": null,
                  "rendered": "Read more...",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="article__hypes"
>
              <div
                            className="icon---default"
              >
                            <img
                                          alt="thumbs-up"
                                          src="thumbs-up.svg"
                            />
                            <div
                                          className="text--small"
                            >
                                          77k
                            </div>
              </div>
              <div>
                            <img
                                          alt="thumbs-up"
                                          src="heart.svg"
                            />
              </div>
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="icon---default"
>
                  <img
                                    alt="thumbs-up"
                                    src="thumbs-up.svg"
                  />
                  <div
                                    className="text--small"
                  >
                                    77k
                  </div>
</div>,
                <div>
                  <img
                                    alt="thumbs-up"
                                    src="heart.svg"
                  />
</div>,
              ],
              "className": "article__hypes",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="thumbs-up"
                      src="thumbs-up.svg"
/>,
                    <div
                      className="text--small"
>
                      77k
</div>,
                  ],
                  "className": "icon---default",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "thumbs-up",
                      "src": "thumbs-up.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "77k",
                      "className": "text--small",
                    },
                    "ref": null,
                    "rendered": "77k",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="thumbs-up"
                    src="heart.svg"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "thumbs-up",
                    "src": "heart.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
